src/main/java/com/gargoylesoftware/htmlunit/javascript/host/xml/XMLHttpRequestEventTarget.java

```diff
@@ -37,8 +41,7 @@
 @JsxClass
 public class XMLHttpRequestEventTarget extends EventTarget {
 
-    private Function loadHandler_;
-    private Function errorHandler_;
+    private final Map<String, Function> eventMapping_ = new HashMap<>();
```

Comment:
Is there a real benefit in having a map here instead of separate vars?

Reply:
The main benefit was that I could use `getFunctionForEvent` to easily identify the `on...` method in the `fireJavascriptProgressEvent` method.

Reply:
The main benefit was that I could use `getFunctionForEvent` to easily identify the `on...` method in the `fireJavascriptProgressEvent` method.